### ISSUE TITLE
pod install failed

### ISSUE THREAD

**Issue by RWOverdijk** [CocoaPods/CocoaPods#6107]
### Command

```
/usr/local/bin/pod install
```
### Report
- _What did you do?_
  I followed the instructions here: https://github.com/airbnb/react-native-maps/blob/master/docs/installation.md and got this error when running `pod install`
- _What did you expect to happen?_
  Not sure? A successful install I think. First time pod user :D
- _What happened instead?_
  This.
### Stack

```
   CocoaPods : 1.1.1
        Ruby : ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin16]
    RubyGems : 2.0.14.1
        Host : Mac OS X 10.12 (16A323)
       Xcode : 8.0 (8A218a)
         Git : git version 2.7.4
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 47d042afa7eba8a457ee22e8fec2cc6207d68021
```
### Plugins

```
cocoapods-deintegrate : 1.0.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.1.1
cocoapods-try         : 1.1.0
```
### Podfile

``` ruby
source 'https://github.com/CocoaPods/Specs.git'

platform :ios, '8.0'
use_frameworks!

target 'AirMapsExplorer' do
  pod 'React', path: '../node_modules/react-native', :subspecs => [
    'Core',
    'RCTActionSheet',
    'RCTGeolocation',
    'RCTImage',
    'RCTLinkingIOS',
    'RCTNetwork',
    'RCTSettings',
    'RCTText',
    'RCTVibration',
    'RCTWebSocket'
  ]

  pod 'GoogleMaps'
end
```
### Error

```
Nanaimo::Reader::ParseError - Dictionary ({#<Nanaimo::String:0x007fe8cebade48 @value="isa", @annotation="">=>#<Nanaimo::String:0x007fe8cebadda8 @value="PBXFileReference", @annotation="">, #<Nanaimo::String:0x007fe8cebadce0 @value="lastKnownFileType", @annotation="">=>#<Nanaimo::QuotedString:0x007fe8cebad998 @value="wrapper.pb-project", @annotation="">, #<Nanaimo::String:0x007fe8cebad8d0 @value="name", @annotation="">=>#<Nanaimo::String:0x007fe8cebad830 @value="RCTActionSheet.xcodeproj", @annotation="">, #<Nanaimo::String:0x007fe8cebad6f0 @value="path", @annotation="">=>#<Nanaimo::String:0x007fe8cebad650 @value="../node_modules/react", @annotation="">}) missing ';' after key-value pair (<Nanaimo::String path> = <Nanaimo::String ../node_modules/react>) at index 6060 (got -)
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/reader.rb:251:in `raise_parser_error'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/reader.rb:165:in `parse_dictionary'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/reader.rb:98:in `parse_object'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/reader.rb:159:in `parse_dictionary'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/reader.rb:98:in `parse_object'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/reader.rb:159:in `parse_dictionary'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/reader.rb:98:in `parse_object'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/reader.rb:73:in `parse!'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.0/lib/xcodeproj/plist.rb:27:in `read_from_path'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.0/lib/xcodeproj/project.rb:200:in `initialize_from_file'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.0/lib/xcodeproj/project.rb:102:in `open'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.1/lib/cocoapods/installer/analyzer.rb:855:in `block (2 levels) in inspect_targets_to_integrate'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.1/lib/cocoapods/installer/analyzer.rb:854:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.1/lib/cocoapods/installer/analyzer.rb:854:in `block in inspect_targets_to_integrate'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.1/lib/cocoapods/user_interface.rb:64:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.1/lib/cocoapods/installer/analyzer.rb:849:in `inspect_targets_to_integrate'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.1/lib/cocoapods/installer/analyzer.rb:66:in `analyze'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.1/lib/cocoapods/installer.rb:236:in `analyze'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.1/lib/cocoapods/installer.rb:150:in `block in resolve_dependencies'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.1/lib/cocoapods/user_interface.rb:64:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.1/lib/cocoapods/installer.rb:149:in `resolve_dependencies'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.1/lib/cocoapods/installer.rb:110:in `install!'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.1/lib/cocoapods/command/install.rb:37:in `run'
/Library/Ruby/Gems/2.0.0/gems/claide-1.0.1/lib/claide/command.rb:334:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.1/lib/cocoapods/command.rb:50:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.1/bin/pod:55:in `<top (required)>'
/usr/local/bin/pod:23:in `load'
/usr/local/bin/pod:23:in `<main>'
```

**Comment by RWOverdijk** [comment]
I get the same when I delete the Podfile, and just run pod init:

```
Nanaimo::Reader::ParseError - Dictionary ({#<Nanaimo::String:0x007f8cd13a0f88 @value="isa", @annotation="">=>#<Nanaimo::String:0x007f8cd13a0ee8 @value="PBXFileReference", @annotation="">, #<Nanaimo::String:0x007f8cd13a0e20 @value="lastKnownFileType", @annotation="">=>#<Nanaimo::QuotedString:0x007f8cd13a0ad8 @value="wrapper.pb-project", @annotation="">, #<Nanaimo::String:0x007f8cd13a0a10 @value="name", @annotation="">=>#<Nanaimo::String:0x007f8cd13a08d0 @value="RCTActionSheet.xcodeproj", @annotation="">, #<Nanaimo::String:0x007f8cd13a07b8 @value="path", @annotation="">=>#<Nanaimo::String:0x007f8cd13a0718 @value="../node_modules/react", @annotation="">}) missing ';' after key-value pair (<Nanaimo::String path> = <Nanaimo::String ../node_modules/react>) at index 6060 (got -)
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/reader.rb:251:in `raise_parser_error'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/reader.rb:165:in `parse_dictionary'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/reader.rb:98:in `parse_object'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/reader.rb:159:in `parse_dictionary'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/reader.rb:98:in `parse_object'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/reader.rb:159:in `parse_dictionary'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/reader.rb:98:in `parse_object'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/reader.rb:73:in `parse!'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.0/lib/xcodeproj/plist.rb:27:in `read_from_path'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.0/lib/xcodeproj/project.rb:200:in `initialize_from_file'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.0/lib/xcodeproj/project.rb:102:in `open'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.1/lib/cocoapods/command/init.rb:41:in `validate!'
/Library/Ruby/Gems/2.0.0/gems/claide-1.0.1/lib/claide/command.rb:333:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.1/lib/cocoapods/command.rb:50:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.1/bin/pod:55:in `<top (required)>'
/usr/local/bin/pod:23:in `load'
/usr/local/bin/pod:23:in `<main>'
```

**Comment by RWOverdijk** [comment]
After opening the project in xcode once, it _does_ work. So I'm guessing this issue doesn't belong here.

**Comment by DanToml** [comment]
@RWOverdijk Could you possibly upload a sample project that generates this error? (I don't have react-native setup, but I'm guessing that something is generating a project with an unquoted string, that Xcode will automatically quote)

**Comment by RWOverdijk** [comment]
@DanToml I could, but I can also provide you the steps required to reproduce this.


### Short version
1. Install react native (`npm i react-native -g`). I'm on 0.36.0
2. Create a new project (`react-native init someProject ; cd $_ ; cd ios`)
3. Run `pod init`
4. Error.

### Long version
1. Install react native (`npm i react-native -g`). I'm on 0.36.0
2. Create a new project (`react-native init someProject ; cd $_ ; cd ios`)
3. Install maps (`npm i react-native-maps`)
4. Copy the Podfile from maps (`wget https://raw.githubusercontent.com/airbnb/react-native-maps/master/example/ios/Podfile`)
5. Open up the podfile, and change the name to your project name (someProject)
6. run `pod install`.
7. Error.

**Comment by zinas** [comment]
Getting the exact error as well, and it happened right after starting a new react native project (ie. `The short version` from the above example)

**Comment by zinas** [comment]
@RWOverdijk what do you mean `opening the project in XCode once`? I tried opening it and it didn't fix it, did you do anything else?

**Comment by RWOverdijk** [comment]
@zinas Hm no. I just opened it, run it once and close xcode again. Then it worked for me. Maybe I accidentally did something to make it work but I don't think so.

**Comment by zinas** [comment]
In case anyone else has the same problem: for me downgrading cocoapods from 1.1.0 to 1.0.0 fixed the issue

**Comment by byennen** [comment]
same issue. adding bundler to the root dir so our team is on the correct cocoapods

**Comment by rileybracken** [comment]
@zinas's fix did not work for me.

Does anyone else have a solution for this?

**Comment by byennen** [comment]
Make sure to uninstall all cocoapods gems.

Then gem install cocoapod -v=1.01

**Comment by zarko-via** [comment]
Having the same issues running pod update.
Uninstalled all versions, install again latest 1.1.1 or 1.0.0 doesn't work too.

Maybe a change in Xcode 8.1 tools?! Seems to fail on Xcode pods project creation.

### Error

```
RuntimeError - Cannot write false to an ascii plist
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/writer.rb:63:in `write_object'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/writer.rb:140:in `write_dictionary_key_value_pair'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/xcode_project_writer.rb:41:in `write_dictionary_key_value_pair'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/writer.rb:119:in `block in write_dictionary'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/writer.rb:118:in `each'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/writer.rb:118:in `write_dictionary'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/xcode_project_writer.rb:19:in `write_dictionary'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/writer.rb:55:in `write_object'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/writer.rb:140:in `write_dictionary_key_value_pair'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/xcode_project_writer.rb:41:in `write_dictionary_key_value_pair'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/writer.rb:119:in `block in write_dictionary'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/writer.rb:118:in `each'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/writer.rb:118:in `write_dictionary'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/xcode_project_writer.rb:19:in `write_dictionary'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/writer.rb:55:in `write_object'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/writer.rb:140:in `write_dictionary_key_value_pair'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/xcode_project_writer.rb:41:in `write_dictionary_key_value_pair'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/xcode_project_writer.rb:29:in `block (2 levels) in write_dictionary'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/xcode_project_writer.rb:28:in `each'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/xcode_project_writer.rb:28:in `block in write_dictionary'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/xcode_project_writer.rb:24:in `each'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/xcode_project_writer.rb:24:in `write_dictionary'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/writer.rb:55:in `write_object'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/writer.rb:140:in `write_dictionary_key_value_pair'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/xcode_project_writer.rb:41:in `write_dictionary_key_value_pair'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/writer.rb:119:in `block in write_dictionary'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/writer.rb:118:in `each'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/writer.rb:118:in `write_dictionary'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/xcode_project_writer.rb:19:in `write_dictionary'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/writer.rb:55:in `write_object'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/writer.rb:29:in `write'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.0/lib/xcodeproj/project.rb:353:in `block in save'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.0/lib/xcodeproj/project.rb:353:in `open'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.0/lib/xcodeproj/project.rb:353:in `save'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/installer.rb:696:in `block in write_pod_project'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/user_interface.rb:144:in `message'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/installer.rb:688:in `write_pod_project'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/installer.rb:173:in `block in generate_pods_project'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/user_interface.rb:63:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/installer.rb:167:in `generate_pods_project'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/installer.rb:119:in `install!'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/command/update.rb:60:in `run'
/Library/Ruby/Gems/2.0.0/gems/claide-1.0.1/lib/claide/command.rb:334:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/command.rb:50:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/bin/pod:55:in `<top (required)>'
/usr/local/bin/pod:23:in `load'
/usr/local/bin/pod:23:in `<main>'
```

**Comment by guoxiaoming-LJ** [comment]
I'm having the same problem with Xcode 8.0. Downgrade cocoapods to 1.0.0 didn't fix the problem.   

<pre>
function abspath() { pushd . > /dev/null; if [ -d "$1" ]; then cd "$1"; dirs -l +0; else cd "`dirname \>) at index 1322818 (got $)
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/reader.rb:251:in `raise_parser_error'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/reader.rb:165:in `parse_dictionary'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/reader.rb:98:in `parse_object'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/reader.rb:159:in `parse_dictionary'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/reader.rb:98:in `parse_object'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/reader.rb:159:in `parse_dictionary'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/reader.rb:98:in `parse_object'
/Library/Ruby/Gems/2.0.0/gems/nanaimo-0.1.2/lib/nanaimo/reader.rb:73:in `parse!'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.0/lib/xcodeproj/plist.rb:27:in `read_from_path'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.0/lib/xcodeproj/project.rb:200:in `initialize_from_file'
/Library/Ruby/Gems/2.0.0/gems/xcodeproj-1.4.0/lib/xcodeproj/project.rb:102:in `open'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/installer/analyzer.rb:695:in `block (2 levels) in inspect_targets_to_integrate'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/installer/analyzer.rb:694:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/installer/analyzer.rb:694:in `block in inspect_targets_to_integrate'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/user_interface.rb:63:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/installer/analyzer.rb:689:in `inspect_targets_to_integrate'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/installer/analyzer.rb:66:in `analyze'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/installer.rb:227:in `analyze'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/installer.rb:151:in `block in resolve_dependencies'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/user_interface.rb:63:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/installer.rb:150:in `resolve_dependencies'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/installer.rb:114:in `install!'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/command/install.rb:37:in `run'
/Library/Ruby/Gems/2.0.0/gems/claide-1.0.1/lib/claide/command.rb:334:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/lib/cocoapods/command.rb:50:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.0.0/bin/pod:55:in `<top (required)>'
/usr/local/bin/pod:23:in `load'
/usr/local/bin/pod:23:in `<main>'

### Plugins

```
cocoapods-deintegrate : 1.0.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.1.1
cocoapods-try         : 1.1.0
</pre>

**Comment by RWOverdijk** [comment]
@guoxiaoming-LJ 8.0 is borked anyway. You should update to 8.1

**Comment by guoxiaoming-LJ** [comment]
@RWOverdijk Upgrading to Xcode 8.1 didn't fix that too : (

**Comment by segiddins** [comment]
This should be fixed in Xcodeproj 1.4.1 and Nanaimo 0.2.1, thanks!

**Comment by zarko-via** [comment]
@segiddins thanks, upgrading xcodeproj to 1.4.1 (sudo gem install xcodeproj), did the trick.
